### PR TITLE
qt6 housekeeping

### PR DESCRIPTION
**Pull request by zomfg**
official 6.2 beta finally seems to have serialport and arm64
the remaining blocker is in `UpdatesProcessor.cpp`, I leave that to you

### CODE REVIEW

**Comment by psieg** [comment]
Thank you!

File: Software/build-vars.prf.default [hunk 1]
```diff
@@ -42,8 +42,9 @@ unix:!macx {
 # When using variables, lrelease complains about QMAKE_MAC_SDK not being set
 # You can override QMAKE_MAC_SDK value by setting this variable.
 macx {
-	# for cross compiling
-	# QMAKE_APPLE_DEVICE_ARCHS = arm64 x86_64h x86_64
+	# by default builds a fat binary with x86_64 + x86_64h (+ arm64)
```
Comment (psieg) [inline Software/build-vars.prf.default:45]: Just for my understanding, this makes a fat Prismatik binary. For it to be useful on M1 you still need either fat Qt binaries as well or separate dmgs with x86 and arm64 qt packages respectively, right?

Comment (zomfg) [reply Software/build-vars.prf.default:45]: > Just for my understanding, this makes a fat Prismatik binary.

yes

> For it to be useful on M1 you still need either fat Qt binaries

yes, and 6.2 mac binaries are fat (TBC with brew release)

> separate dmgs with x86 and arm64 qt packages respectively

hmmm I never tried fat binary with slim dependencies... but this shouldn't be an issue here with all-in-one dmg
that being said you can manually merge/strip architectures so you still can get to fat-everything (which is what i used for 5.15 M1 previews)

on a side note about that snippet: it does not work the same with pre-6, in 5.15 `QMAKE_APPLE_DEVICE_ARCHS` is defined by default in a common qmake spec so the condition in `build-vars` is not met and it builds with that default arch
so if 5.15 is also fatish (x86_64 + x86_64h), Prismatik won't be unless you override
you can undefine `QMAKE_APPLE_DEVICE_ARCHS` before including custom `build-vars` but that would "break" the cli `QMAKE_APPLE_DEVICE_ARCHS=blah qmake...`
so I left as is
in 6  it's not defined before and defaults to something only when you don't define it so it works out

Comment (psieg) [reply Software/build-vars.prf.default:45]: Yeah if Qt had separate binaries I'd have voted for slim Prismatik as well.
Thanks, I'm not at all familiar with mac nor Qt development ;)

**Pull request merged**